internal/savedsearches/manager.go

```diff
@@ -0,0 +1,295 @@
+		writeStatus(w, http.StatusBadRequest)
+		return
+	}
+
+	dbSavedSearch := m.DbClient.GetSavedSearchById(savedSearchId)
+	eligibilityIds := getEligibilityIdsFromDbSavedSearches([]*db.SavedSearch{dbSavedSearch})
+	dbEligibilities := m.DbClient.GetEligibilitiesByIDs(eligibilityIds)
+	categoryIds := getCategoryIdsFromDbSavedSearches([]*db.SavedSearch{dbSavedSearch})
+	dbCategories := m.DbClient.GetCategoriesByIDs(categoryIds)
+
+	eligibilityIdToName := make(map[int]string)
+	for _, dbEligibility := range dbEligibilities {
+		eligibilityIdToName[dbEligibility.Id] = dbEligibility.Name.String
+	}
+	categoryIdToName := make(map[int]string)
+	for _, dbCategory := range dbCategories {
+		categoryIdToName[dbCategory.Id] = dbCategory.Name
+	}
+
+	response := FromDBType(dbSavedSearch, eligibilityIdToName, categoryIdToName)
+
+	writeJson(w, response)
+}
+
+// Delete saved search by ID
+// not done
+// (note - I don't have the user auth stuff here)
+//
+//	@Summary		Delete saved search by ID
+//	@Description	delete a saved search for user
+//	@Tags			saved_searches
+//	@Accept			json
+//	@Produce		json
+//	@Success		200	{object}	savedsearches.SavedSearch
+//	@Router			/saved_searches/{id} [delete]
+func (m *Manager) Delete(w http.ResponseWriter, r *http.Request) {
+	id, err := strconv.Atoi(chi.URLParam(r, "id"))
+	if err != nil {
+		log.Printf("%v", err)
+	}
+	err = m.DbClient.DeleteSavedSearchById(id)
+	if err != nil {
+		log.Print(err)
+		writeStatus(w, http.StatusInternalServerError)
+	}
+
+	writeStatus(w, http.StatusOK)
+}
+
+func writeJson(w http.ResponseWriter, object interface{}) {
+	output, err := json.Marshal(object)
+	if err != nil {
+		fmt.Println("error:", err)
+	}
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	_, err = w.Write(output)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func writeStatus(w http.ResponseWriter, responseStatus int) {
+	w.WriteHeader(responseStatus)
+}
+
+// Create a flattened, uniquified list of eligibility IDs.
+func getEligibilityIdsFromDbSavedSearches(dbSavedSearches []*db.SavedSearch) []int {
+	eligibilityIdMap := make(map[int]bool)
+	for _, dbSavedSearch := range dbSavedSearches {
+		for _, id := range dbSavedSearch.Search.Eligibilities {
+			eligibilityIdMap[id] = true
+		}
+	}
+	var eligibilityIds []int
+	for id := range eligibilityIdMap {
+		eligibilityIds = append(eligibilityIds, id)
+	}
+	return eligibilityIds
+}
+
+// Create a flattened, uniquified list of category IDs.
+func getCategoryIdsFromDbSavedSearches(dbSavedSearches []*db.SavedSearch) []int {
+	categoryIdMap := make(map[int]bool)
+	for _, dbSavedSearch := range dbSavedSearches {
+		for _, id := range dbSavedSearch.Search.Categories {
+			categoryIdMap[id] = true
+		}
+	}
+	var categoryIds []int
+	for id := range categoryIdMap {
+		categoryIds = append(categoryIds, id)
+	}
+	return categoryIds
+}
+
+// Return the elements of a that are not in b
+func diffStringSlices(a []string, b []string) []string {
```

Comment:
maybe not-"go like" but feels like we would want to eventually make a shared_logic with a manager for something like this?? or like this would be built-in with soemthing?

Reply:
We can move this to a common utility file, theres also a `slice.Contains` function that might make this easier https://pkg.go.dev/slices